### PR TITLE
add xgboost library

### PR DESCRIPTION
**Pull request by rootfs**
Once the pkg is built, on the Kepler CI, run `xgboost.sh  --skip-license  --prefix=/usr/local && ldconfig` to install xgboost package and library

### CODE REVIEW

File: .github/workflows/publish_xgboost.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+          export XGBOOST_TAG=$(cat xgboost_tag)
+          sudo apt-get update -y
+          # ubuntu 20.04 ships with cmake 3.16, but xgboost needs 3.18+. Install cmake from pip
+          sudo apt-get install git make g++ python3-pip -y
+          sudo pip install cmake --upgrade
+          git clone --recursive https://github.com/dmlc/xgboost.git
+          cd xgboost && git checkout tags/v${XGBOOST_TAG} && mkdir -p build && cd build && cmake .. && make -j4 && make package && cp xgboost-${XGBOOST_TAG}-Linux.sh /tmp/xgboost.sh
+    - if: startsWith(github.event.ref, 'refs/tags/')
+      name: Release
+      uses: softprops/action-gh-release@master
```
Comment (SamYuan1990) [inline .github/workflows/publish_xgboost.yml]: can we release file in two gha?

Comment (SamYuan1990) [reply .github/workflows/publish_xgboost.yml]: seems conflicts with https://github.com/sustainable-computing-io/kepler-ci-artifacts/blob/main/.github/workflows/publish_bcc_deb.yml#L51, can we have a test before merge?

Comment (rootfs) [reply .github/workflows/publish_xgboost.yml]: Since they are in different GHA, they should be released separately.

Comment (SamYuan1990) [reply .github/workflows/publish_xgboost.yml]: @rootfs , I suppose as https://github.com/softprops/action-gh-release will only create one release page per once time. Which means, for each time, there will be two release. One for bcc and one for xgboost?

Comment (SamYuan1990) [reply .github/workflows/publish_xgboost.yml]: further question as do we just build xgboost for CI usage but not our final build?
if yes, we can just update current GHA with xgboost, as we just run this GHA for one time and for CI or internal usage only.

**Review by SamYuan1990 — APPROVED** [review]
Consider we will remove bcc soon, I can give a LTGM with comments for this PR. if we were luck, we can have xgboost in release page with current code.

File: .github/workflows/publish_xgboost.yml (hunk 1 above)
Comment (rootfs) [reply .github/workflows/publish_xgboost.yml]: fixed

**Comment by rootfs** [comment]
feedback addressed, let's merge it and get the xgboost library

**Pull request merged**